src/Http/Request.php

```diff
@@ -293,7 +298,17 @@ public function getRemoteHost()
 	 */
 	public function getRawBody()
 	{
-		return $this->rawBodyCallback ? call_user_func($this->rawBodyCallback) : NULL;
+		return $this->rawBodyCallback ? call_user_func($this->rawBodyCallback, $this) : NULL;
```

Comment:
You're using the same callback, that also parses the JSON for application/json, aren't you?